runtime.go

```diff
@@ -320,6 +320,10 @@ func (r *runtime) InstantiateModule(
 		return
 	}
 
+	if closeNotifier, ok := ctx.Value(internalclose.NotifierKey{}).(internalclose.Notifier); ok {
```

Comment:
note: this will notify on host and guest modules, so if you use the same context you can get multiple notifications on runtime.Close. I added this note to the godoc as well.